kobo/worker/main.py

```diff
@@ -8,7 +8,7 @@
 import logging
 
 # IMPORTANT: import taskd first to set os.environ["PROJECT_DEFAULT_CONFIG_FILE"]
-from kobo.worker.taskmanager import TaskManager, TaskContainer
+from kobo.worker.taskmanager import TaskManager
```

Comment:
These changes aren't backwards-compatible, because you're reducing the set of symbols exported by these modules.

The pylint (or whatever) tool is only telling you that these imports are unused in *this file*.  But this is a library, and the tool is not able to tell you whether the imports are used from elsewhere.

I made some effort to look into it myself (grepping through sources of projects using kobo).  There's no entirely robust way of doing it but I was able to confirm at least that this specific change *will* break some other code.

Example 1, in kobo itself:

```
kobo/kobo/admin/templates/worker/__project_name__:import kobo.worker.main
kobo/kobo/admin/templates/worker/__project_name__:    kobo.worker.main.TaskContainer.register_module(kobo.worker.tasks, prefix="task_")
```

That kobo.worker.main.TaskContainer would have worked before this change, and now it'd be broken.  Note also that this is the template which new applications will use, so that kobo.worker.main.TaskContainer reference probably made its way into several projects by now.

Example 2, in pubd:

```
rcm-pub/pubd/__init__.py:from kobo.worker.main import *
rcm-pub/pubd/__init__.py:TaskContainer.register_module(pubd.tasks)
```

Again, this change would make TaskContainer undefined and break this code.

I'm making this comment on just one of these changes, but same thing applies to most of the others here.

I think we should value backwards-compatibility and trouble-free upgrades of kobo library for our users more than satisfying this pylint check, so I think this cleanup is a bad idea, at least for the kobo and django related imports.

I don't mind the cleanup for the core modules as I don't believe anyone would have written code like "from django.auth.models import re".

Reply:
I added this one again, not sure about others if they are a problem.